app/representers/metric_representer.rb

```diff
@@ -1,10 +1,12 @@
+# frozen_string_literal: true
+
 module MetricRepresenter
   include ThreeScale::JSONRepresenter
 
   wraps_resource
 
   property :id
-  property :name
+  property :name, if: ->(*) { !backend_api_metric? }
```

Comment:
Does the following work?

```suggestion
  property :name, unless: ->(*) { backend_api_metric? }
```

Yet, why removing it anyway? Wouldn't it be better to keep the same response no matter if the metric is a sercie metric or a backend api metric?

Reply:
@guicassolato 
The unless does not work unfortunately. It is the first thing I tried.

About the `name`, it is deprecated since 2014 and we should remove it from everywhere, but as that is risky right now, at a bare minimum let's don't add it to new endpoints.
Btw the `name` is actually the same as the `system_name`.

See this:
https://github.com/3scale/porta/blob/679f351d2382e677f557c602ad73b7122c905efd/app/controllers/admin/api/metrics_base_controller.rb#L17

https://github.com/3scale/porta/blob/679f351d2382e677f557c602ad73b7122c905efd/app/models/metric.rb#L1

https://github.com/3scale/porta/blob/679f351d2382e677f557c602ad73b7122c905efd/app/models/metric.rb#L135